apps/meteor/server/services/upload/service.ts

```diff
@@ -0,0 +1,39 @@
+import type { IMessage, IUpload } from '@rocket.chat/core-typings';
+import { Meteor } from 'meteor/meteor';
+
+import { ServiceClassInternal } from '../../sdk/types/ServiceClass';
+import type {
+	ISendFileLivechatMessageParams,
+	ISendFileMessageParams,
+	IUploadFileParams,
+	IUploadService,
+} from '../../sdk/types/IUploadService';
+import { FileUpload } from '../../../app/file-upload/server';
+
+export class UploadService extends ServiceClassInternal implements IUploadService {
+	protected name = 'upload';
+
+	async uploadFile({ buffer, details }: IUploadFileParams): Promise<IUpload> {
+		const fileStore = FileUpload.getStore('Uploads');
+		return fileStore.insert(details, buffer);
+	}
+
+	async sendFileMessage({ roomId, file, userId, message }: ISendFileMessageParams): Promise<IMessage | undefined> {
+		return Meteor.runAsUser(userId, () => Meteor.call('sendFileMessage', roomId, null, file, message));
+	}
+
+	async sendFileLivechatMessage({ roomId, visitorToken, file, message }: ISendFileLivechatMessageParams): Promise<IMessage> {
+		return Meteor.call('sendFileLivechatMessage', roomId, visitorToken, file, message);
+	}
```

Comment:
Don't want to sound like one of those service purists, but, should we have this "sendmessage" actions on this service? It's easier yeah, but 🤔 @sampaiodiego

Reply:
I think so..

just not sure if this service should also save the message, or just somehow associate the file with the message.

Reply:
I'd say it would be good for the future to have a "messaging" service and do those actions there. 🤔 I'll resolve this one, but will create tasks for updating this in the future.
Maybe using this call to proxy messages to reduce changes